### PR TITLE
Region map

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/image_processing_effects.c [hunk 1]
```diff
@@ -0,0 +1,1224 @@
...
+static void SetPresetPalette_GrayscaleSmall(void);
+static void QuantizePalette_GrayscaleSmall(void);
+static void SetPresetPalette_BlackAndWhite(void);
+static void QuantizePalette_BlackAndWhite(void);
+static u16 QuantizePixel_Standard(u16*);
+static u16 QuantizePixel_GrayscaleSmall(u16*);
+static u16 QuantizePixel_Grayscale(u16*);
+static u16 QuantizePixel_PrimaryColors(u16*);
+
+extern const u8 gPointillismPoints[][3];
```
Comment (jiangzhengwenjz) [inline src/image_processing_effects.c]: Are we going to move the array to here?

File: src/image_processing_effects.c [hunk 2]
```diff
@@ -0,0 +1,1224 @@
...
+                *pixel = RGB2(grayValue, grayValue, grayValue);
+            }
+        }
+    }
+}
+
+static void ApplyImageEffect_Pointillism(void)
+{
+    u32 i;
+    for (i = 0; i < 3200; i++)
```
Comment (jiangzhengwenjz) [inline src/image_processing_effects.c]: `3200` looks like `NELEMS(gPointillismPoints)` if the array's moved here

File: src/region_map.c [hunk 3]
```diff
@@ -0,0 +1,4402 @@
...
+static u8 sub_80C3878(u8 a0);
+static u8 sub_80C3AC8(u8 a0);
+static void GetPlayerPositionOnRegionMap_HandleOverrides(void);
+static u8 GetSelectedMapSection(u8 a0, u8 a1, s16 a2, s16 a3);
+static void sub_80C41D8(u16 a0, u16 a1);
+static void sub_80C4244(void);
+static void sub_80C4324(bool8 a0);
+static void sub_80C4348(void);
+static u16 sub_80C4380();
+static u16 sub_80C438C();
```
Comment (jiangzhengwenjz) [inline src/region_map.c]: Missing arg type

File: src/region_map.c [hunk 4]
```diff
@@ -0,0 +1,4402 @@
...
+    u8 mapType;
+    bool8 regionMapPermissions[4];
+    u8 field_479B;
+    u8 field_479C;
+    u8 filler_479D[3];
+    u8 field_47A0;
+    u8 filler_47A1[3];
+    u8 field_47A4;
+    u8 filler_47A5[3];
+    u8 field_47A8;
```
Comment (jiangzhengwenjz) [inline src/region_map.c]: maybe it's because the fields following are aligned to 4?

File: src/region_map.c [hunk 5]
```diff
@@ -0,0 +1,4402 @@
...
+        MAPSEC_RIXY_CHAMBER,
+        MAPSEC_VIAPOIS_CHAMBER,
+        MAPSEC_NONE
+    }
+};
+
+ALIGNED(4) const bool8 sRegionMapPermissions[3][4] = {
+    {TRUE , TRUE , TRUE , FALSE},
+    {FALSE, FALSE, FALSE, FALSE},
+    {FALSE, FALSE, FALSE, TRUE }
```
Comment (jiangzhengwenjz) [inline src/region_map.c]: space following `TRUE`

Comment (PikalaxALT) [reply src/region_map.c]: intentional for visual alignment

**Pull request merged**